src/main/java/stsjorbsmod/actions/GetCardFromDrawPileAction.java

```diff
@@ -0,0 +1,37 @@
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+
+public class GetCardFromDrawPileAction extends AbstractGameAction {
+    private AbstractCard card;
+
+    public GetCardFromDrawPileAction(AbstractCard card) {
+        this.actionType = ActionType.CARD_MANIPULATION;
+        this.card = card;
+        this.duration = Settings.ACTION_DUR_FAST;
+    }
+
+    public void update() {
+        if (this.duration == Settings.ACTION_DUR_FAST) {
+            if (AbstractDungeon.player.drawPile.contains(this.card) && AbstractDungeon.player.hand.size() < 10) {
+                AbstractDungeon.player.hand.addToHand(this.card);
+                this.card.unhover();
+                this.card.setAngle(0.0F, true);
+                this.card.lighten(false);
+                this.card.drawScale = 0.12F;
+                this.card.targetDrawScale = 0.75F;
+                this.card.applyPowers();
+                AbstractDungeon.player.drawPile.removeCard(this.card);
+            }
+
+            AbstractDungeon.player.hand.refreshHandLayout();
+            AbstractDungeon.player.hand.glowCheck();
+        }
+
+        this.tickDuration();
+        this.isDone = true;
```

Comment:
`isDone = true;` makes `tickDuration()` and the initial `if(this.duration ...)` check both unnecessary.